ksp_plugin/flight_plan.cpp

```diff
@@ -0,0 +1,179 @@
+﻿#include "ksp_plugin/flight_plan.hpp"
+
+namespace principia {
+namespace ksp_plugin {
+
+FlightPlan::FlightPlan(
+    not_null<DiscreteTrajectory<Barycentric>*> root,
+    Instant const& initial_time,
+    Instant const& final_time,
+    Mass const& initial_mass,
+    not_null<Ephemeris<Barycentric>*> ephemeris,
+    AdaptiveStepSizeIntegrator<
+        Ephemeris<Barycentric>::NewtonianMotionEquation> const& integrator,
+    Length const& length_integration_tolerance,
+    Speed const& speed_integration_tolerance)
+    : initial_time_(initial_time),
+      final_time_(final_time),
+      initial_mass_(initial_mass),
+      ephemeris_(ephemeris),
+      integrator_(integrator),
+      length_integration_tolerance_(length_integration_tolerance),
+      speed_integration_tolerance_(speed_integration_tolerance) {
+  CHECK(final_time_ >= initial_time_);
+  //ephemeris_->Prolong(final_time_);
+  auto it = root->LowerBound(initial_time);
+  if (it.time() != initial_time_) {
+    --it;
+  }
+  segments_.emplace(root->NewForkWithCopy(it.time()));
+  ResetLastSegment();  // TODO(phl): A NewForkWithoutCopy would be nicer.
+  CoastLastSegment(final_time_);
+}
+
+int FlightPlan::size() const {
+  return manœuvres_.size();
+}
+
+NavigationManœuvre const& FlightPlan::Get(int index) {
+  CHECK_LE(0, index);
+  CHECK_LT(index, size());
+  return manœuvres_[index];
+}
+
+bool FlightPlan::Append(Burn burn) {
+  auto manœuvre =
```

Comment:
Make the type explicit, it's unclear what's happening here.  Also, const and Append a copy.